### ISSUE TITLE
[Question/Request] Custom Page

### ISSUE THREAD

**Issue by kjellherzke** [sveltia/sveltia-cms#180]
Hi! I recently came across sveltia-cms when searching for a better alternative to decap cms and I am very satisfied with this decision.

Nevertheless, one main reason for me to switch was the difficulty of developing an extension, mainly to the cms's ui. More specifically, my objective is to be able to add custom pages to the sveltia-cms's admin dashboard/ui, but I am wondering about which way of implementation.

I could fork this repository and modify the source code directly, but I would like to know if there is any other way of doing so that you would suggest me instead? Furthermore, I would be interested in knowing whether or not do you (_natively_) support _custom pages_?

**Comment by kyoshino** [comment]
Hey @kjellherzke, thanks for checking out Sveltia CMS! It’s a good idea to have handy custom pages on the CMS. But I think I can work on it soonish, as Static CMS compatibility is planned [after the 1.0 release](https://github.com/sveltia/sveltia-cms?tab=readme-ov-file#after-the-10-release), hopefully early next year.

Technically speaking, I need to figure out how to integrate React with Svelte-based Sveltia CMS, which is also required for custom widgets. Alternatively, I could just add support for pages inside iframes.

Anyway, I’ll put this on my todo list!

**Comment by kjellherzke** [comment]
Hey @kyoshino, I do understand that. Nevertheless I am questioning why you wouldn't want to primarily focus on Svelte-based extensions (as this would also initially be easier to implement). Furthermore, the final idea for me is that the cms is actually rather part of an admin dashboard, then all of its content, if you understand me correctly, although the cms is of course the actual provider of the admin ui, it should be extendible.

**Comment by kyoshino** [comment]
Integrating Svelte is rather hard, because the code has to be compiled. Compiling Svelte code on the fly is [possible](https://svelte.dev/docs/svelte-compiler), but it’s a heavy process. I’m instead thinking about supporting [Preact](https://preactjs.com/)+[HTM](https://github.com/developit/htm) as I said [somewhere else](https://github.com/sveltia/sveltia-cms/discussions/153), which is easier to develop and run without a build process. And for backward compatibility with Netlify/Decap CMS, I need to implement React integration anyway.

I’ll prepare developer docs after 1.0, because Sveltia CMS currently does not accept pull requests and I haven’t implemented the [Test backend](https://decapcms.org/docs/test-backend/) yet. That said, here’s what I’m doing to develop Sveltia CMS:

1. Run Sveltia CMS with `pnpm dev` on `localhost:5173`.
2. Run a SvelteKit site project with a Sveltia CMS configuration  on `localhost:5174`.
3. Open `localhost:5173` with your browser.

If you’re using a framework other than SvelteKit, you need to specify the site URL with the `VITE_SITE_URL` environment variable. You may also need to set up CORS. [Here is an example for e11y](https://gist.github.com/kyoshino/3636f0ab6f857968248aad5269b0f75d).

**Comment by kyoshino** [comment]
And I forgot to say: at this time, it’s possible to embed Sveltia CMS within your project using a [custom mount element](https://decapcms.org/docs/custom-mounting/) instead of embedding a custom page in the CMS.

**Comment by kyoshino** [comment]
By the way, Sveltia CMS is currently migrating to Svelte 5. It’s a massive rewrite with over 2,500 lines of changes. If you want to extend the CMS, you might want to wait until it’s done next month.

**Comment by kyoshino** [comment]
Static CMS is dead as of today 😢 The documentation is archived at https://staticjscms.netlify.app/docs/additional-links

**Comment by CarloBu** [comment]
> Static CMS is dead as of today 😢 The documentation is archived at https://staticjscms.netlify.app/docs/additional-links

yeah, I saw in the discord the message. I didn't use it personally, but it's sad that there are one less git-based cms.

**Comment by kyoshino** [comment]
I’ve also never used it for my projects, but they did great job IMO. I like their documentation.

BTW do you want me to create a Discord server? Here we go: https://discord.gg/6zC5eTCw 🦀

**Comment by CarloBu** [comment]
> I’ve also never used it for my projects, but they did great job IMO. I like their documentation.
> 
> BTW do you want me to create a Discord server? Here we go: https://discord.gg/6zC5eTCw 🦀

yeah, it's sad... now you can steal their documentation 🫣

new discord. noice!

**Comment by RDjarbeng** [comment]
Is the discord limited to those invited only?

**Comment by kyoshino** [comment]
It has expired by mistake. A new invitation link is https://discord.gg/5hwCGqup5b

**Comment by RDjarbeng** [comment]
Thanks, joined now

**Comment by errnerr** [comment]
Just wanted comment to add my +1 to this request. It would be cool to embed web analytics on a custom page. Excellent work you've done so far with the CMS. It's amazing!

**Comment by 0UserName** [comment]
So, is it now possible to integrate a custom page into the CMS? Some kind of API (even an internal one). In particular, I am interested in the possibility of adding a page within which a table could be rendered based on data obtained from a remote API. And perhaps access to authentication data would be required, namely a PAT (Personal Access Token) in the context of GitHub.

**Comment by kyoshino** [comment]
There is no API or whatever yet, but as [I said earlier](https://github.com/sveltia/sveltia-cms/issues/180#issuecomment-2256151443), you can rather embed the CMS on your page.

**Comment by 0UserName** [comment]
Let's say the CMS is embedded in a custom page, is it possible to somehow use the authentication data?

**Comment by kyoshino** [comment]
Your access token is stored in the browser’s local storage, an item named `sveltia-cms.user`. You can use the `token` property in the stringified object (or override it if you want).
